kafka-connect-cloud-common/src/test/scala/io/lenses/streamreactor/connect/cloud/common/sink/naming/CloudKeyNamerTest.scala

```diff
@@ -69,6 +70,30 @@ class CloudKeyNamerTest extends AnyFunSuite with Matchers with OptionValues with
 
   private val s3KeyNamer = CloudKeyNamer(formatSelection, partitionSelection, fileNamer, paddingService)
 
+  test("the partition values do not replace / or \\ characters") {
+    val partitionSelection =
+      PartitionSelection(isCustom = false, List(HeaderPartitionField(PartitionNamePath("h"))), Values)
+    val keyNamer = CloudKeyNamer(formatSelection, partitionSelection, fileNamer, paddingService)
+    val either: Either[SinkError, Map[PartitionField, String]] = keyNamer.processPartitionValues(
+      MessageDetail(
+        NullSinkData(None),
+        NullSinkData(None),
+        Map("h" -> StringSinkData("val1/val2")),
+        None,
+        topicPartition.topic,
+        topicPartition.partition,
+        topicPartition.offset,
+      ),
+      topicPartition.toTopicPartition,
+    )
+
+    either match {
+      case Left(_) => fail("Should not have failed")
+      case Right(value) =>
+        value shouldBe Map(HeaderPartitionField(PartitionNamePath("h")) -> "val1/val2")
+    }
```

Comment:
I would probably just use `EitherValues` trait which is already being used

`either.value should be Map(HeaderPartitionField(PartitionNamePath("h")) -> "val1/val2")

Then no need for the conditional match 😁